frontends/mit-open/src/page-components/Header/MenuButton.tsx

```diff
@@ -0,0 +1,44 @@
+import { Button, styled } from "ol-components"
+import MenuIcon from "@mui/icons-material/Menu"
+import React from "react"
+
+const StyledMenuIcon = styled(MenuIcon)(({ theme }) => ({
+  color: theme.custom.colors.darkGray1,
+}))
+
+const MenuButtonText = styled.div(({ theme }) => ({
+  alignSelf: "center",
+  color: theme.custom.colors.darkGray2,
+  paddingLeft: "16px",
+  textTransform: "none",
+  ...theme.typography.subtitle2,
+}))
+
+const MenuButtonInner = styled.div({
+  display: "flex",
+  padding: "8px 24px 8px 0px",
+  alignItems: "flex-start",
+})
+
+const StyledMenuButton = styled(Button)({
+  background: "transparent",
+  "&:hover:not(:disabled)": {
+    background: "transparent",
+  },
+})
+
+interface MenuButtonProps {
+  text: string
+  onClick: React.MouseEventHandler<HTMLButtonElement> | undefined
+}
+
+const MenuButton: React.FC<MenuButtonProps> = ({ text, onClick }) => (
+  <StyledMenuButton onClick={onClick}>
+    <MenuButtonInner>
+      <StyledMenuIcon />
```

Comment:
None of them seemed to really match the style. Plus, when I go to make the header updates this button is going to need to have logic to change the icon based on the state of the drawer so it will be good to isolate that logic to its own component.

Reply:
(I'm still a bit torn on the name though)